internal/test/e2e/setup.go

```diff
@@ -81,6 +84,11 @@ func (e *E2ESession) setup(regex string) error {
 	logger.V(1).Info("Instance created", "instance-id", instanceId)
 	e.instanceId = instanceId
 
+	err = ec2.TagInstance(e.session, instanceId, jobIdTagKey, e.parentJobId)
```

Comment:
I would vote to accept a `map[string]string` in `CreateInstance` so the instance is created with the tag from the beginning.